azurerm/internal/services/media/media_streaming_policy_resource.go

```diff
@@ -0,0 +1,693 @@
+	if len(input) == 0 {
+		return nil
+	}
+
+	playReady := input[0].(map[string]interface{})
+
+	customLicenseURLTemplate := ""
+	if v := playReady["custom_license_acquisition_url_template"]; v != nil {
+		customLicenseURLTemplate = v.(string)
+	}
+
+	customAttributes := ""
+	if v := playReady["custom_attributes"]; v != nil {
+		customAttributes = v.(string)
+	}
+
+	return &media.StreamingPolicyPlayReadyConfiguration{
+		CustomLicenseAcquisitionURLTemplate: utils.String(customLicenseURLTemplate),
+		PlayReadyCustomAttributes:           utils.String(customAttributes),
+	}
+}
+
+func expandDefaultKey(input []interface{}) *media.DefaultKey {
+	if len(input) == 0 {
+		return nil
+	}
+
+	defaultKey := input[0].(map[string]interface{})
+	defaultKeyResult := &media.DefaultKey{}
+
+	if v := defaultKey["policy_name"]; v != nil {
+		defaultKeyResult.PolicyName = utils.String(v.(string))
+	}
+
+	if v := defaultKey["label"]; v != nil {
+		defaultKeyResult.Label = utils.String(v.(string))
+	}
+
+	return defaultKeyResult
+}
+
+func expandFairPlay(input []interface{}) *media.StreamingPolicyFairPlayConfiguration {
+	if len(input) == 0 {
+		return nil
+	}
+
+	fairPlay := input[0].(map[string]interface{})
+
+	customLicenseURLTemplate := ""
+	if v := fairPlay["custom_license_acquisition_url_template"]; v != nil {
+		customLicenseURLTemplate = v.(string)
+	}
+
+	allowPersistentLicense := false
+	if v := fairPlay["allow_persistent_license"]; v != nil {
+		allowPersistentLicense = v.(bool)
+	}
+
+	return &media.StreamingPolicyFairPlayConfiguration{
+		CustomLicenseAcquisitionURLTemplate: utils.String(customLicenseURLTemplate),
+		AllowPersistentLicense:              utils.Bool(allowPersistentLicense),
+	}
+}
+
+func flattenNoEncryption(input *media.NoEncryption) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	return flattenEnabledProtocols(input.EnabledProtocols)
+}
+
+func flattenEnabledProtocols(input *media.EnabledProtocols) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	dash := false
+	if input.Dash != nil {
+		dash = *input.Dash
+	}
+
+	download := false
+	if input.Download != nil {
+		download = *input.Download
+	}
+
+	hls := false
+	if input.Hls != nil {
+		hls = *input.Hls
+	}
+
+	smoothStreaming := false
+	if input.SmoothStreaming != nil {
+		smoothStreaming = *input.SmoothStreaming
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"dash":             dash,
+			"download":         download,
+			"hls":              hls,
+			"smooth_streaming": smoothStreaming,
+		},
+	}
+}
+
+func flattenCommonEncryptionCenc(input *media.CommonEncryptionCenc) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	enabledProtocols := make([]interface{}, 0)
+	if input.EnabledProtocols != nil {
+		enabledProtocols = flattenEnabledProtocols(input.EnabledProtocols)
+	}
+
+	widevineTemplate := ""
+	if input.Drm != nil && input.Drm.Widevine != nil {
```

Comment:
there's a panic on the next line here if `input.Drm.Widevine.CustomLicenseAcquisitionURLTemplate` is nil, so could we nil-check that too:

```suggestion
	if input.Drm != nil && input.Drm.Widevine != nil && input.Drm.Widevine.CustomLicenseAcquisitionURLTemplate != nil {
```